jvcl/run/JvPcx.pas

```diff
@@ -72,6 +74,21 @@ implementation
 uses
   JvResources;
 
+class function TJvPcx.CanLoadFromStream(Stream: TStream): Boolean;
+var
+  Header: TPcxHeader;
+  P: Int64;
+begin
+  P := Stream.Position;
+
+  try
+    Result := (Stream.Read(Header, SizeOf(Header)) = SizeOf(Header)) and
+      (Header.Id = $0A) and (Header.BytesPerLine mod 2 = 0);
+  finally
+    Stream.Position := P;
```

Comment:
Some streams cannot "roll back", like the compression streams. In that case, they trigger an exception which would make it impossible to use the class in that context.
I'm not sure it would be possible to work around this, maybe a look at the PNG classes from the VCL itself would help.